client/database.go

```diff
@@ -107,3 +80,55 @@ func (db *database) CreateCollections(ctx context.Context, dbName string, schema
 
 	return tx.Commit(ctx)
 }
+
+func GetCollection[T Model](db *database, model interface{}) *Collection[T] {
```

Comment:
why not have this method on the database object?

Reply:
Limited Go generics support doesn't allow typed methods.
So we cannot do:
```
func (db *database) Collection[T Model]() *Collection[T]
```
https://github.com/golang/go/issues/49085

```
Although methods of a generic type may use the type's parameters, methods may not themselves have additional type parameters. Where it would be useful to add type arguments to a method, people will have to write a suitably parameterized top-level function.
```